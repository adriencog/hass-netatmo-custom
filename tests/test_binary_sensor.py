"""Support for Netatmo binary sensors."""

from unittest.mock import AsyncMock

import pytest
from syrupy import SnapshotAssertion

from homeassistant.components.netatmo.const import NETATMO_EVENT
from homeassistant.const import CONF_WEBHOOK_ID, Platform
from homeassistant.core import HomeAssistant
from homeassistant.helpers import entity_registry as er

from .common import selected_platforms, simulate_webhook, snapshot_platform_entities

from tests.common import MockConfigEntry, async_capture_events


@pytest.mark.usefixtures("entity_registry_enabled_by_default")
async def test_entity(
    hass: HomeAssistant,
    config_entry: MockConfigEntry,
    netatmo_auth: AsyncMock,
    snapshot: SnapshotAssertion,
    entity_registry: er.EntityRegistry,
) -> None:
    """Test entities."""
    await snapshot_platform_entities(
        hass,
        config_entry,
        Platform.BINARY_SENSOR,
        entity_registry,
        snapshot,
    )


async def test_webhook_siren_event(
    hass: HomeAssistant, config_entry: MockConfigEntry, netatmo_auth: AsyncMock
) -> None:
    """Test that person events are handled."""
    with selected_platforms([Platform.BINARY_SENSOR]):
        assert await hass.config_entries.async_setup(config_entry.entry_id)

        await hass.async_block_till_done()

    test_netatmo_event = async_capture_events(hass, NETATMO_EVENT)
    assert not test_netatmo_event

    fake_webhook_event = {
        "event_type": "home_alarm",
        "camera_id": "12:34:56:00:f1:62",
        "device_id": "12:34:56:00:f1:62",
        "module_id": "12:34:56:00:e3",
        "event_id": "1234567890",
        "message": "Update Siren",
        "push_type": "NACamerahome_alarm",
    }

    webhook_id = config_entry.data[CONF_WEBHOOK_ID]
    await simulate_webhook(hass, webhook_id, fake_webhook_event)
    assert test_netatmo_event


async def test_webhook_opening_event(
    hass: HomeAssistant, config_entry: MockConfigEntry, netatmo_auth: AsyncMock
) -> None:
    """Test window events are handled."""
    with selected_platforms([Platform.BINARY_SENSOR]):
        assert await hass.config_entries.async_setup(config_entry.entry_id)

        await hass.async_block_till_done()

    assert hass.states.get("binary_sensor.window_hall_window").state == "off"
    test_netatmo_event = async_capture_events(hass, NETATMO_EVENT)
    assert not test_netatmo_event

    fake_webhook_event = {
        "event_type": "tag_open",
        "camera_id": "12:34:56:00:f1:62",
        "device_id": "12:34:56:00:f1:62",
        "module_id": "12:34:56:00:86:99",
        "event_id": "1234567890",
        "message": "Window open",
        "push_type": "NACamera-tag_open",
    }

    webhook_id = config_entry.data[CONF_WEBHOOK_ID]
    await simulate_webhook(hass, webhook_id, fake_webhook_event)
    assert hass.states.get("binary_sensor.window_hall_window").state == "on"

    assert test_netatmo_event
