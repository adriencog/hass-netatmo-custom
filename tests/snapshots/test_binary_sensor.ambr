# serializer version: 1
# name: test_entity[binary_sensor.baby_bedroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.baby_bedroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:26:68:92-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.baby_bedroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Baby Bedroom Connectivity',
      'latitude': 13.377726,
      'longitude': 52.516263,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.baby_bedroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.bedroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.bedroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:26:69:0c-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.bedroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Bedroom Connectivity',
      'latitude': 13.377726,
      'longitude': 52.516263,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.bedroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_entity[binary_sensor.kitchen_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.kitchen_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:25:cf:a8-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.kitchen_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Kitchen Connectivity',
      'latitude': 13.377726,
      'longitude': 52.516263,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.kitchen_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.livingroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.livingroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:26:65:14-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.livingroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Livingroom Connectivity',
      'latitude': 13.377726,
      'longitude': 52.516263,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.livingroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.parents_bedroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.parents_bedroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:3e:c5:46-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.parents_bedroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Parents Bedroom Connectivity',
      'latitude': 13.377726,
      'longitude': 52.516263,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.parents_bedroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.sirene_in_hall_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'reachable',
    'unique_id': '12:34:56:00:e3:9b-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Sirene in hall Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.sirene_in_hall_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_safety-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.sirene_in_hall_safety',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.SAFETY: 'safety'>,
    'original_icon': None,
    'original_name': 'Safety',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'monitoring',
    'unique_id': '12:34:56:00:e3:9b-monitoring',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_safety-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'safety',
      'friendly_name': 'Sirene in hall Safety',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.sirene_in_hall_safety',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_sound-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.sirene_in_hall_sound',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.SOUND: 'sound'>,
    'original_icon': None,
    'original_name': 'Sound',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'status',
    'unique_id': '12:34:56:00:e3:9b-sound',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.sirene_in_hall_sound-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'sound',
      'friendly_name': 'Sirene in hall Sound',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.sirene_in_hall_sound',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_entity[binary_sensor.villa_bathroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_bathroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:80:7e:18-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_bathroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Bathroom Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_bathroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.villa_bedroom_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_bedroom_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:80:44:92-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_bedroom_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Bedroom Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_bedroom_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.villa_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:80:bb:26-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Connectivity',
      'latitude': 46.123456,
      'longitude': 6.1234567,
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.villa_garden_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_garden_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:03:1b:e4-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_garden_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Garden Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_garden_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.villa_outdoor_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_outdoor_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:80:1c:42-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_outdoor_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Outdoor Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_outdoor_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_entity[binary_sensor.villa_rain_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.villa_rain_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': None,
    'unique_id': '12:34:56:80:c1:ea-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.villa_rain_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Villa Rain Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.villa_rain_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.window_hall_connectivity-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.window_hall_connectivity',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.CONNECTIVITY: 'connectivity'>,
    'original_icon': None,
    'original_name': 'Connectivity',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'reachable',
    'unique_id': '12:34:56:00:86:99-reachable',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.window_hall_connectivity-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'connectivity',
      'friendly_name': 'Window Hall Connectivity',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.window_hall_connectivity',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'on',
  })
# ---
# name: test_entity[binary_sensor.window_hall_motion-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.window_hall_motion',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.MOTION: 'motion'>,
    'original_icon': None,
    'original_name': 'Motion',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'status',
    'unique_id': '12:34:56:00:86:99-motion',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.window_hall_motion-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'motion',
      'friendly_name': 'Window Hall Motion',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.window_hall_motion',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_entity[binary_sensor.window_hall_vibration-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.window_hall_vibration',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.VIBRATION: 'vibration'>,
    'original_icon': None,
    'original_name': 'Vibration',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'status',
    'unique_id': '12:34:56:00:86:99-vibration',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.window_hall_vibration-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'vibration',
      'friendly_name': 'Window Hall Vibration',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.window_hall_vibration',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
# name: test_entity[binary_sensor.window_hall_window-entry]
  EntityRegistryEntrySnapshot({
    'aliases': set({
    }),
    'area_id': None,
    'capabilities': None,
    'config_entry_id': <ANY>,
    'device_class': None,
    'device_id': <ANY>,
    'disabled_by': None,
    'domain': 'binary_sensor',
    'entity_category': None,
    'entity_id': 'binary_sensor.window_hall_window',
    'has_entity_name': True,
    'hidden_by': None,
    'icon': None,
    'id': <ANY>,
    'labels': set({
    }),
    'name': None,
    'options': dict({
    }),
    'original_device_class': <BinarySensorDeviceClass.WINDOW: 'window'>,
    'original_icon': None,
    'original_name': 'Window',
    'platform': 'netatmo',
    'previous_unique_id': None,
    'supported_features': 0,
    'translation_key': 'status',
    'unique_id': '12:34:56:00:86:99-opening',
    'unit_of_measurement': None,
  })
# ---
# name: test_entity[binary_sensor.window_hall_window-state]
  StateSnapshot({
    'attributes': ReadOnlyDict({
      'attribution': 'Data provided by Netatmo',
      'device_class': 'window',
      'friendly_name': 'Window Hall Window',
    }),
    'context': <ANY>,
    'entity_id': 'binary_sensor.window_hall_window',
    'last_changed': <ANY>,
    'last_reported': <ANY>,
    'last_updated': <ANY>,
    'state': 'off',
  })
# ---
